src/items/associated-items.md

```diff
@@ -293,8 +293,8 @@ type that the definition has to implement.
 An *associated constant definition* defines a constant associated with a
 type. It is written the same as a [constant item].
 
-Unlike [free] constants, associated constant definitions undergo
-[constant evaluation] only when referenced.
+Associated constant definitions undergo [constant evaluation] only when
+referenced.
```

Comment:
And only after monomorphization. Nothing is actually done on generic constants, even if we could attempt it nowadays

Reply:
I pushed [d0b33a2](https://github.com/rust-lang/reference/pull/1147/commits/d0b33a2990c305e8793a2e53dbc77b6edec1b018) and [37e3c15](https://github.com/rust-lang/reference/pull/1147/commits/37e3c15c52b31f3cda9fd359d3bb66a3e69bb5ff) to try to explain this.